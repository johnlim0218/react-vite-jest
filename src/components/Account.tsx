import * as React from 'react';

interface AccountProps {
  id: string;
  password: string;
}

function Account ({ id, password }: AccountProps) {
  return (
    <div>
      <h2>{id}</h2>
      <h2>{password}</h2>
    </div>
  )
}

export default Account;