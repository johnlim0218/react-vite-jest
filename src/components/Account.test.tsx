import * as React from 'react';
import { render } from "@testing-library/react";
import Account from "./Account";

it ("matches snapshot", () => {
  const utils = render(<Account id="gwanwoo.john" password="1234" />)
  expect(utils.container).toMatchSnapshot()
})

it ("shows the props correctly", () => {
  const utils = render(<Account id="gwanwoo.john" password="1234" />)
  utils.getByText("gwanwoo.john")
  utils.getByText("1234")
})