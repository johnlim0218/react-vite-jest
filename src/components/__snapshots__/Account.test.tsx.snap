// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <h2>
      gwanwoo.john
    </h2>
    <h2>
      1234
    </h2>
  </div>
</div>
`;
